src/main/java/com/loofah/graph/api/queries/SkillsQuery.java

```diff
@@ -21,7 +26,27 @@ public SkillsQuery(final SkillRepository skillRepository) {
 
     @Override
     public List<Skill> get(final DataFetchingEnvironment dataFetchingEnvironment) {
-        return skillRepository.findAll();
+
+        SkillFilter skillFilter = objectMapper.convertValue(dataFetchingEnvironment.getArgument("filter"), SkillFilter.class);
+
+        Query query = new Query();
+
+        if(skillFilter != null) {
```

Comment:
Nitpick: as an idea, we could remove some of the nesting in this conditional logic by using optionals. e.g. something like:

```
Optional<SkillFilter> skillFilter = Optional.ofNullable(skillFilter);

skillFilter.map(filter::getCategoryId).ifPresent( {{add the categoryId to the query}} )
skillFilter.map(filter::getGradeId).ifPresent( {{add the gradeId to the query}} )
```

No worries if you find it more readable as is though :)

Reply:
good suggestion, have updated